### PR TITLE
fix(canned responses): handle canned response moving up in the list

### PR DESCRIPTION
**Pull request by ajohn25**
## Description

This changes the handling for dragging canned responses to better accomodate the case where a canned response is dragged up in the list instead of down.

## Motivation and Context

The case where a canned response is dragged up in the list doesn't consistently work with the current handling.

## How Has This Been Tested?
This has been tested locally

## Screenshots (if appropriate):

https://github.com/politics-rewired/Spoke/assets/76596635/b2b528a0-ecae-498d-9ec3-8e8e177a878e

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have included updates for the documentation accordingly.

### CODE REVIEW

File: src/containers/AdminCampaignEdit/sections/CampaignCannedResponsesForm/index.tsx [hunk 1]
```diff
@@ -133,7 +133,7 @@ class CampaignCannedResponsesForm extends React.Component<InnerProps, State> {
     const cannedResponsesToAdd = this.state.cannedResponsesToAdd.concat({
       ...response,
       id: newId,
-      displayOrder: cannedResponses.length + 1
+      displayOrder: cannedResponses.length
```
Comment (henryk1229) [inline src/containers/AdminCampaignEdit/sections/CampaignCannedResponsesForm/index.tsx:136]: question: does this result in two `cannedResponses` with the same display order?

**Review by henryk1229 — APPROVED** [review]
lgtm

File: src/containers/AdminCampaignEdit/sections/CampaignCannedResponsesForm/index.tsx (hunk 1 above)
Comment (ajohn25) [reply src/containers/AdminCampaignEdit/sections/CampaignCannedResponsesForm/index.tsx:136]: We moved from 1-based indexing to 0-based indexing after feedback in #1540 so this was a holdover from that which was really throwing me off while debugging for a bit 😭

**Comment by ajohn25** [comment]
Git's showing commits in the wrong order 🤔 the most recent checks passed

**Pull request merged**